File: 2315-count-asterisks/2315-count-asterisks.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        pipe = 0
        asterisks = 0
        i = 0
        while i < len(s) :
            if s[i] == '|' :
                pipe += 1
            if s[i] == '*' and pipe % 2 == 0 :
                asterisks += 1 
            i += 1
        return asterisks

File: 2315-count-asterisks/2315-count-asterisks.js
/**
 * @param {string} s
 * @return {number}
 */
var countAsterisks = function(s) {
    let pipe = 0;
    let asterisks = 0;
    for(let i = 0; i < s.length; i++){
        if(s[i] === '|'){
            pipe++
        }else if(s[i] === '*' && pipe % 2 === 0){
            asterisks++
        }
    }
    return asterisks;
};